### ISSUE TITLE
TestFileSD is flakey

### ISSUE THREAD

**Issue by brian-brazil** [prometheus/prometheus#1001]
I'm seeing failures like:

```
--- FAIL: TestFileSD (0.15s)
        file_test.go:89: Unexpected empty target group received: fixtures/_test.json:0
        file_test.go:89: Unexpected empty target group received: fixtures/_test.json:1
FAIL
```

maybe 50% of the time but I can't pin down what's causing them.

**Comment by juliusv** [comment]
Got similar errors sometimes:

```
ok  	_/home/julius/gosrc/src/github.com/prometheus/prometheus/retrieval	3.760s
time="2015-08-18T14:45:01+02:00" level=error msg="Error reading file \"fixtures/_test.yml\": yaml: did not find expected node content" file=file.go line=171 
time="2015-08-18T14:45:01+02:00" level=error msg="Error reading file \"fixtures/_test.yml\": yaml: did not find expected node content" file=file.go line=171 
time="2015-08-18T14:45:01+02:00" level=error msg="Error reading file \"fixtures/_test.yml\": yaml: did not find expected node content" file=file.go line=171 
time="2015-08-18T14:45:01+02:00" level=error msg="Error reading file \"fixtures/_test.json\": invalid character ']' looking for beginning of value" file=file.go line=171 
time="2015-08-18T14:45:01+02:00" level=error msg="Error reading file \"fixtures/_test.json\": invalid character ']' looking for beginning of value" file=file.go line=171 
time="2015-08-18T14:45:01+02:00" level=error msg="Error reading file \"fixtures/_test.json\": invalid character ']' looking for beginning of value" file=file.go line=171 
--- FAIL: TestFileSD (0.15s)
	file_test.go:89: Unexpected empty target group received: fixtures/_test.json:0
	file_test.go:89: Unexpected empty target group received: fixtures/_test.json:1
FAIL
FAIL	_/home/julius/gosrc/src/github.com/prometheus/prometheus/retrieval/discovery	0.178s
```

**Comment by juliusv** [comment]
Ok, I guess the extra logged errors above are intentional and just show up because the test fails for other reasons.

**Comment by fabxc** [comment]
Yes

**Comment by brian-brazil** [comment]
Fixed by #1030
